src/renderers/shaders/ShaderChunk/shadowmap_pars_fragment.glsl

```diff
@@ -16,4 +16,136 @@
 
 	}
 
+	#if defined(POINT_LIGHT_SHADOWS)
+
+		float unpack1K ( vec4 color ) {
+		
+			const vec4 bitSh = vec4( 1.0 / (256.0 * 256.0 * 256.0), 1.0 / (256.0 * 256.0), 1.0 / 256.0, 1.0 );
+			return dot( color, bitSh ) * 1000.0;
+			
+		}
+
+		/**
+		*  cubeToUV() maps a 3D direction vector suitable for cube texture mapping to a 2D
+		*  vector suitable for 2D texture mapping. This code uses the following layout for the
+		*  2D texture:
+		*
+		*  Y y
+		*  xzXZ
+		*
+		*  Y - Positive y direction
+		*  y - Negative y direction
+		*  X - Positive x direction
+		*  x - Negative x direction
+		*  Z - Positive z direction
+		*  z - Negative z direction
+		*
+		*  Alternate code for different layouts, more compact code arrangement, and seam
+		*  skipping can be found here: https://gist.github.com/tschw/da10c43c467ce8afd0c4
+		*/
+
+		vec2 cubeToUV( vec3 v, float texelSizeX, float texelSizeY ) {
+
+			// Horizontal cross layout:
+			const vec2 Squares = vec2( 4.0, 2.0 );
+			const vec2 Center = vec2( 1.0, 0.0 );
+
+			// Size of a square in UV space:
+			const vec2 TexSquareSize = 1.0 / Squares;
+
+			// UV space offset of the center of the center square of the cross:
+			const vec2 TexCoordOffs = TexSquareSize * ( 0.5 + Center );
+
+			// Factors to scale square space (-1..+1 per square) to UV space:
+			const vec2 TexSquareScale = TexSquareSize * 0.5;
+
+			// Just less than a texel in square space when divided by resolution:
+			const float TexEps = 1.5; // = min(Squares.x, Squares.y) - 0.5;
+
+			vec3 absV = abs( v );
+			vec3 sgnV = sign( v );
+
+			// Intersect unit cube
+
+			float scale = 1.0 / max( absV.x, max( absV.y, absV.z ) );
+
+			v *= scale; 
+			absV *= scale;
+
+			// Determine gate factors
+
+			// gate.? is one when on left / right, bottom / top, back
+			float eps = TexEps * texelSizeY;
+			vec3 gate = step( 1.0 - eps, vec3( absV.xy, v.z ) );
+
+			// prefer any square over bottom / top
+			float notX = 1. - gate.x;
+			float notZ = 1. - gate.z;
+			gate.y *= notX * notZ;
+			// prefer back over side
+			gate.x *= notZ;
+
+			// Unwrap
+
+			// start with xy coordinates
+			vec2 planar = v.xy;
+
+			// stop at the last texel (can use a factor of 1.0 for NEAREST)
+			float yTexelSize = 2.0 * Squares.y * texelSizeY;
```

Comment:
> Calculating lots of stuff that is cancelled out plus doing additional stuff to avoid branches is actually worse than what we'd get from compiling a few ifs for a target that can't jump. The latter implementation takes fewer instructions in total!

> Furthermore, many GPUs can jump and use an SIMT execution model, IOW they really skip code that is not needed by any of the (typically 32) lock-stepping threads in the same execution vector processing nearby fragments. Hardware of this type benefits most from the branching variant.

Given this, I think we should use your updated code. I replaced the existing code with your implementation of cubeToUVCompact() and it works fine. I will include that in the next update to the PR.

Are there any other issues I should address before updating the PR?